### PR TITLE
source environment hooks in topological order in prefix level local_setup files

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ament/ament_package#7

http://ci.ros2.org/job/ros2_batch_ci_linux/479/
http://ci.ros2.org/job/ros2_batch_ci_osx/381/
http://ci.ros2.org/job/ros2_batch_ci_windows/531/

### CODE REVIEW

File: ament_package/template/prefix_level/_order_packages.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        pkg_name = pkg_name_without_deps[0]
+        to_be_ordered.remove(pkg_name)
+        ordered.append(pkg_name)
+        # remove item from dependency lists
+        for k in list(run_dependencies.keys()):
+            if pkg_name in run_dependencies[k]:
+                run_dependencies[k].remove(pkg_name)
+
+    for name in ordered:
+        print(name)
```
Comment (wjwwood) [inline ament_package/template/prefix_level/_order_packages.py]: Debug output?

Comment (wjwwood) [inline ament_package/template/prefix_level/_order_packages.py]: Nvm, I think this is for communicating the package names.

**Comment by wjwwood** [comment]
It doesn't seem to have impacted the build time, based on comparing the previous builds and the CI jobs you linked. lgtm

**Comment by dirk-thomas** [comment]
I think the impact is way to small to be noticeable in our current CI jobs. Locally for me it is around 50ms to invoke the Python script (instead of using a hard coded ordered list).

**Comment by dirk-thomas** [comment]
And remove executable bits from all templates.

**Pull request merged**